haystack/document_store/milvus.py

```diff
@@ -0,0 +1,384 @@
+        self._create_collection_and_index_if_not_exist(self.index)
+        self.return_embedding = return_embedding
+        self.embedding_field = embedding_field
+
+        super().__init__(
+            url=sql_url,
+            update_existing_documents=update_existing_documents,
+            index=index
+        )
+
+    def __del__(self):
+        return self.milvus_server.close()
+
+    def _create_collection_and_index_if_not_exist(
+        self,
+        index: Optional[str] = None,
+        index_param: Optional[Dict[str, Any]] = None
+    ):
+        index = index or self.index
+        index_param = index_param or self.index_param
+
+        status, ok = self.milvus_server.has_collection(collection_name=index)
+        if not ok:
+            collection_param = {
+                'collection_name': index,
+                'dimension': self.vector_dim,
+                'index_file_size': self.index_file_size,
+                'metric_type': self.metric_type
+            }
+
+            self.milvus_server.create_collection(collection_param)
+
+            self.milvus_server.create_index(index, self.index_type, index_param)
+
+    def _create_document_field_map(self) -> Dict:
+        return {
+            self.index: self.embedding_field,
+        }
+
+    def write_documents(
+            self, documents: Union[List[dict], List[Document]], index: Optional[str] = None, batch_size: int = 10_000
+    ):
+        """
+        Add new documents to the DocumentStore.
+
+        :param documents: List of `Dicts` or List of `Documents`. If they already contain the embeddings, we'll index
+                                  them right away in FAISS. If not, you can later call update_embeddings() to create & index them.
+        :param index: (SQL) index name for storing the docs and metadata
+        :param batch_size: When working with large number of documents, batching can help reduce memory footprint.
+        :return:
+        """
+        index = index or self.index
+        self._create_collection_and_index_if_not_exist(index)
+        field_map = self._create_document_field_map()
+        document_objects = [Document.from_dict(d, field_map=field_map) if isinstance(d, dict) else d for d in documents]
+
+        add_vectors = False if document_objects[0].embedding is None else True
+
+        for i in range(0, len(document_objects), batch_size):
+            vector_ids = []
+            if add_vectors:
+                doc_ids = []
+                embeddings = []
+                for doc in document_objects[i: i + batch_size]:
+                    doc_ids.append(doc.id)
+                    if isinstance(doc.embedding, np.ndarray):
+                        embeddings.append(doc.embedding.tolist())
+                    elif isinstance(doc.embedding, list):
+                        embeddings.append(doc.embedding)
+                    else:
+                        raise AttributeError("Document embedded in unrecognized format")
```

Comment:
Let's make this error message more descriptive: `(f"Format of supplied document embedding ({type(doc.embedding}) is not supported. Please use list or numpy.ndarray")`